### PR TITLE
Replace old database queries with the new ones

### PR DESCRIPTION
**Pull request by annando**
There are still many left - for a future PR.

### CODE REVIEW

File: mod/message.php [hunk 1]
```diff
@@ -148,11 +148,7 @@ function message_content(App $a)
 
 		$cmd = $a->argv[1];
 		if ($cmd === 'drop') {
-			$r = q("DELETE FROM `mail` WHERE `id` = %d AND `uid` = %d LIMIT 1",
-				intval($a->argv[2]),
-				intval(local_user())
-			);
-			if ($r) {
+			if (dba::delete('mail', ['id' => $a->argv[2]])) {
```
Comment (MrPetovan) [inline mod/message.php]: There's no previous ownership test, I believe the `uid` condition should be kept in place.

File: mod/settings.php [hunk 2]
```diff
@@ -149,9 +149,7 @@ function settings_post(App $a)
 		check_form_security_token_redirectOnErr('/settings/oauth', 'settings_oauth');
 
 		$key = $_POST['remove'];
-		q("DELETE FROM tokens WHERE id='%s' AND uid=%d",
-			dbesc($key),
-			local_user());
+		dba::delete('tokens', ['id' => $key]);
```
Comment (MrPetovan) [inline mod/settings.php]: There's no previous ownership test, I believe the `uid` condition should be kept in place.

File: src/Model/GContact.php [hunk 3]
```diff
@@ -239,10 +239,8 @@ public static function sanitize($gcontact)
 
 			if ($alternate && ($gcontact['network'] == NETWORK_OSTATUS)) {
 				// Delete the old entry - if it exists
-				$r = q("SELECT `id` FROM `gcontact` WHERE `nurl` = '%s'", dbesc(normalise_link($orig_profile)));
-				if (DBM::is_result($r)) {
-					q("DELETE FROM `gcontact` WHERE `nurl` = '%s'", dbesc(normalise_link($orig_profile)));
-					q("DELETE FROM `glink` WHERE `gcid` = %d", intval($r[0]["id"]));
```
Comment (MrPetovan) [inline src/Model/GContact.php]: Is this deletion now triggered by the cascading delete?

File: src/Protocol/DFRN.php [hunk 4]
```diff
@@ -2078,94 +2078,30 @@ private static function processRelocation($xpath, $relocation, $importer)
...
-			intval($importer["id"]),
-			intval($importer["importer_uid"]),
-			dbesc(normalise_link($old["url"]))
-		);
+		$fields = ['name' => $relocate["name"], 'avatar' => $relocate["avatar"],
+			'url' => $relocate["url"], 'nurl' => normalise_link($relocate["url"]),
+			'addr' => $relocate["addr"], 'request' => $relocate["request"],
+			'confirm' => $relocate["confirm"], 'notify' => $relocate["notify"],
+			'poll' => $relocate["poll"], 'site-pubkey' => $relocate["sitepubkey"]];
+		$condition = ["(`id` = ?) OR (`nurl` = ?)", $importer["id"], normalise_link($old["url"])];
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php]: This variable isn't used later in the code.

File: src/Model/GContact.php (hunk 3 above)
Comment (annando) [reply src/Model/GContact.php]: Yes, it is. (I checked)

File: src/Protocol/DFRN.php (hunk 4 above)
Comment (annando) [reply src/Protocol/DFRN.php]: Well, I forgot the database command.

**Pull request merged**